src/python/python/dict_to_xml.py

```diff
@@ -0,0 +1,615 @@
+from collections import OrderedDict
+import numpy as np
+import os
+from shutil import copy2
+import math
+
+class Files:
+    MAIN = 0
+    MATS = 1
+    GEOM = 2
+    EMIT = 3
+    CAMS = 4
+    #TODO: Volumes
+
+class WriteXML:
+    '''
+    File Writing API
+    '''
+
+    defaults = { #arg name: default name
+        'sample_count': 'spp',
+        'width': 'resx',
+        'height': 'resy'
+    }
+
+    def __init__(self, path, split_files=False):
+        from mitsuba.core import PluginManager
+        self.pmgr = PluginManager.instance()
+        self.split_files = split_files
+        self.scene_data = [OrderedDict([('type','scene')]), #MAIN
+                      OrderedDict(), #MATS
+                      OrderedDict(), #GEOM
+                      OrderedDict(), #EMIT
+                      OrderedDict()] #CAMS
+        self.com_count = 0 #counter for comment ids
+        self.exported_ids = set()
+        self.copy_count = {'tex': 0, 'mesh': 0, 'spectrum': 0}#counters for giving unique names to copied files
+        self.copied_paths = {}
+        self.files = []
+        self.file_names = [] #relative paths to the fragment files
+        self.file_tabs = []
+        self.file_stack = []
+        self.current_file = Files.MAIN
+        self.directory = ''
+        self.set_filename(path)
+
+    def data_add(self, key, value, file=Files.MAIN):
+        self.scene_data[file].update([(key, value)])
+
+    def add_comment(self, comment, file=Files.MAIN):
+        key = "__com__%d" % self.com_count
+        self.com_count += 1
+        self.data_add(key,{'type':'comment', 'value': comment}, file)
+
+    def add_include(self, file):
+        key = "__include__%d" % file
+        value = {'type':'include', 'filename':self.file_names[file]}
+        self.data_add(key, value) #add to main
+
+    def wf(self, ind, st, tabs=0):
+        '''
+        ind                 int
+        st                  string
+        tabs                int
+
+        Write a string to file index ind.
+        Optionally indent the string by a number of tabs
+
+        Returns None
+        '''
+
+        if len(self.files) == 0:
+            scene = object()
+            scene.name = 'untitled'
+            scene.frame_current = 1
+            self.set_filename(scene, 'default')
```

Comment:
Should this check happen everytime we try to write a line in the files?

Reply:
This is a relic of the XML writing routine of the old addon that I forgot to remove. I don't think it is needed at all.